### PR TITLE
Implement 14 cards in CORE cardset

### PR DESCRIPTION
**Pull request by utilForever**
This revision includes:
- Implement 14 cards in CORE cardset
    - Divine Spirit (CS2_236)
    - Starving Buzzard (CS2_237)
    - Houndmaster (DS1_070)
    - Timber Wolf (DS1_175)
    - Tundra Rhino (DS1_178)
    - Tracking (DS1_184)
    - Shattered Sun Cleric (EX1_019)
    - Warsong Commander (EX1_084)
    - Totemic Might (EX1_244)
    - Mortal Coil (EX1_302)
    - Gurubashi Berserker (EX1_399)
    - Murloc Tidehunter (EX1_506)
    - Grimscale Oracle (EX1_508)
    - Animal Companion (NEW1_031)
- Create some tasks
    - `FilterStackTask`
    - `GetGameTagTask`
    - `MathSubTask`
    - `RandomEntourageTask`
- Change logic of some tasks
    - `ChooseTask`
    - `ConditionTask`
    - `ControlTask`
    - `SummonTask`
- Correct LGTM issues (1 C++ issue)
- Add new aura type: `AuraType::FIELD`
- Add new trigger type: `TriggerType::SUMMON` and `TriggerType::TAKE_DAMAGE`
- Add new self condition: `IsDead`, `IsRace` and `IsTagValue`
- Add new targeting play requirement: `PlayReq::REQ_NUM_MINION_SLOTS`

### CODE REVIEW

File: Includes/Rosetta/Models/Battlefield.hpp [hunk 1]
```diff
@@ -61,14 +61,10 @@ class Battlefield
     //! \return The position of minion in battlefield.
     std::optional<std::size_t> FindMinionPos(Minion& minion);
```
Comment (CreatorSeraph) [inline Includes/Rosetta/Models/Battlefield.hpp]: 수정이 없었던 부분이긴 하지만 const를 적용할수 있을것 같습니다

File: Sources/Rosetta/Tasks/SimpleTasks/SummonTask.cpp [hunk 2]
```diff
@@ -31,20 +32,63 @@ TaskID SummonTask::GetTaskID() const
...
+        switch (m_side)
+        {
+            case SummonSide::DEFAULT:
+                summonPos = -1;
+                break;
+            case SummonSide::RIGHT:
+            {
+                auto field = m_source->owner->GetField();
+                const auto minion = dynamic_cast<Minion*>(m_source);
+                auto fieldPos = field.FindMinionPos(*minion);
```
Comment (CreatorSeraph) [inline Sources/Rosetta/Tasks/SimpleTasks/SummonTask.cpp]: BattleField의 FindMinionPos함수가 const라면 field와 fieldPos변수도 const로 만들수 있을것같습니다
단순 소환위치만 잡아주는변수기 때문에 const사용도 고려해보는것이 어떨까요?

File: Includes/Rosetta/Models/Battlefield.hpp (hunk 1 above)
Comment (utilForever) [reply Includes/Rosetta/Models/Battlefield.hpp]: 수정했습니다.

File: Sources/Rosetta/Tasks/SimpleTasks/SummonTask.cpp (hunk 2 above)
Comment (utilForever) [reply Sources/Rosetta/Tasks/SimpleTasks/SummonTask.cpp]: 수정했습니다.

**Pull request merged**